### ISSUE TITLE
Unable to get htlengine to work with minimum testcase

### ISSUE THREAD

**Issue by alexanderdegre** [adobe/htlengine#23]
Hi there,

I'm unable to get htlengine working. By following the base instructions in the readme (running cli.js against simple2) I receive the following error:

```
$ node src/cli.js test/simple2.html
module.js:549
    throw err;
    ^

Error: Cannot find module '@adobe/htlengine/src/runtime/Runtime'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/aledeg/Development/htlengine/out.js:15:17)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
```

So I thought okay that makes sense, the generated out-file is meant to be used from somewhere where htlengine is a dependency. So I created a minimum-possible testcase which uses the same setup/architecture as in these test files ( you can find this here: https://github.com/alexanderdegre/htlengine_test ) which fails with the following error:

```
/Users/aledeg/Development/htlengine_test/index.js:12
    return service(resource);
           ^

TypeError: service is not a function
    at doThing (/Users/aledeg/Development/htlengine_test/index.js:12:12)
    at Object.<anonymous> (/Users/aledeg/Development/htlengine_test/index.js:18:14)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
```

Interestingly, when I modify htlengine's test files (cli.js, out.js etc) so that way the test data and template are "hardcoded" like in my own testcase I also end up with exactly the same error.

Also interestingly, despite this, running `yarn test` from htlengine's root folder passes successfully.

I'm not sure where or how my testcase or my environment differs that prevents my output JS from being evaluated and turned into html. Perhaps somebody here can offer some insight?

Also which versions of the following are you using? Perhaps I can adjust to the same versions and give it another try.
```
nvm version: 0.33.4
node version: 10.10.0
npm version: 6.4.1
antlr version: stable 4.7.1 (via brew)
```

Thank you so much for the efforts!

**Comment by tripodsan** [comment]
without looking into too much detail, I think the `out.js` has a `module.exports.main = ...`. so you're code probably look like: `service.main(resource)`.

I will fix the readme/example soon. thanks for reporting!

**Comment by alexanderdegre** [comment]
I looked deeper into how 'out.js' is built / intended to work and was able to make some small changes in my test repo minimum testcase to generate html output.

It turns out that the provided generator method in the example needs to be slightly different depending on whether or not the `includeRuntime` is set to true. With the runtime, a `main()` is exported and expects a resource argument. Without the runtime, `exports` is simply given a function and expects an argument of `new Runtime()`.

Some of my confusion came from how the examples look in combination with the few comments of "todo: async support" in `Compiler.js`. After digging into `Runtime.js` I learned that what's actually being returned is a `co()` method aka async using Promises. So the example code breaks in that it looks like it should directly output a plain html string,

(  meaning specifically the following code in `cli.js`:  )

```
const html = engine(resource, template);

// eslint-disable-next-line no-console
console.log(html);
```

but actually a Promise is delivered and expects you to use `.then((result) => { ... }` to get a meaningful result.

I've modified my own test repo with the small change needed to actually output html. I could additionally open a small PR for this example change if you like 🙂, but I believe it would still be incomplete. I have the feeling the examples were made in a time when the engine worked quite differently internally
